doc/perception_pipeline/src/detect_and_add_cylinder_collision_object_demo.cpp

```diff
@@ -47,21 +47,20 @@
 
 class CylinderSegment
 {
+  ros::NodeHandle nh_;
+  moveit::planning_interface::PlanningSceneInterface planning_scene_interface_;
+  ros::Subscriber cloud_subscriber_;
+
 public:
   CylinderSegment()
+    : cloud_subscriber_(nh_.subscribe("/camera/depth_registered/points", 1, &CylinderSegment::cloudCB, this))
   {
-    ros::NodeHandle nh;
-    // Initialize subscriber to the raw point cloud
-    ros::Subscriber sub = nh.subscribe("/camera/depth_registered/points", 1, &CylinderSegment::cloudCB, this);
-    // Spin
-    ros::spin();
   }
 
   /** \brief Given the parameters of the cylinder add the cylinder to the planning scene.
-      @param cylinder_params - Pointer to the struct AddCylinderParams. */
+      @param cylinder_params - struct AddCylinderParams. */
```

Comment:
this function has no param, why is it listed here in doxygen?

Reply:
Because of late undocumented refactoring by the student back then. Addressed